clients/js-legacy/src/actions.ts

```diff
@@ -0,0 +1,272 @@
+import type { ConfirmOptions, Connection, Signer } from '@solana/web3.js';
+import { PublicKey, sendAndConfirmTransaction, SystemProgram, Transaction } from '@solana/web3.js';
+import {
+    RECORD_PROGRAM_ID,
+    RECORD_META_DATA_SIZE,
+    RECORD_CHUNK_SIZE_PRE_INITIALIZE,
+    RECORD_CHUNK_SIZE_POST_INITIALIZE,
+} from './constants';
+import {
+    createInitializeInstruction,
+    createWriteInstruction,
+    createSetAuthorityInstruction,
+    createCloseAccountInstruction,
+    createReallocateInstruction,
+} from './instructions';
+
+/**
+ * Initialize a record account
+ *
+ * @param connection      Connection to use
+ * @param payer           Payer of the transaction fees
+ * @param record          Record account address
+ * @param authority       Record account authority
+ * @param recordSize      Size of the record to be stored
+ * @param confirmOptions  Options for confirming the transaction
+ *
+ * @return Signature of the confirmed transaction
+ */
+export async function initializeRecord(
+    connection: Connection,
+    payer: Signer,
+    record: Signer,
+    authority: PublicKey,
+    recordSize: number,
+    confirmOptions?: ConfirmOptions,
+) {
+    const programId = RECORD_PROGRAM_ID;
+    const recordAccountSize = RECORD_META_DATA_SIZE + recordSize;
+    const lamports = await connection.getMinimumBalanceForRentExemption(Number(recordAccountSize));
+    const transaction = new Transaction().add(
+        SystemProgram.createAccount({
+            fromPubkey: payer.publicKey,
+            newAccountPubkey: record.publicKey,
+            space: Number(recordAccountSize),
+            lamports,
+            programId,
+        }),
+        createInitializeInstruction(
+            record.publicKey,
+            authority,
+        )
+    );
+    return await sendAndConfirmTransaction(connection, transaction, [payer, record], confirmOptions);
+}
+
+/**
+ * Write to an initialized record account
+ *
+ * @param connection      Connection to use
+ * @param payer           Payer of the transaction fees
+ * @param record          Record account address
+ * @param authority       Record account authority keypair
+ * @param recordSize      Offset to store the record bytes
+ * @param buffer          Record bytes
+ * @param confirmOptions  Options for confirming the transaction
+ *
+ * @return Signature of the confirmed transaction
```

Comment:
nit: this is returning a list of signatures, right?

Reply:
Fixed in [2e4646c](https://github.com/solana-program/record/pull/20/commits/2e4646c1f72100323b3875ed5864ced434ece5ae)